lib/pacemaker/pcmk_sched_instances.c

```diff
@@ -740,6 +740,43 @@ pcmk__find_compatible_instance(const pe_resource_t *match_rsc,
     return instance;
 }
 
+/*!
+ * \internal
+ * \brief Unassign an instance if ordering without interleave match is mandatory
+ *
+ * \param[in] first          'First' action in an ordering
+ * \param[in] then           'Then' action in an ordering
+ * \param[in] then_instance  'Then' instance that has no interleave match
+ * \param[in] type           Group of enum pe_ordering flags to apply
+ * \param[in] current        If true, "then" action is stopped or demoted
+ *
+ * \return true if \p then_instance was unassigned, otherwise false
+ */
+static bool
+unassign_if_mandatory(pe_action_t *first, pe_action_t *then,
+                      pe_resource_t *then_instance, uint32_t type, bool current)
+{
+    // Allow "then" instance to go down even without an interleave match
+    if (current) {
+        pe_rsc_trace(then->rsc,
+                     "%s has no instance to order before stopping "
+                     "or demoting %s",
+                     first->rsc->id, then_instance->id);
+
+    /* If the "first" action must be runnable, but there is no "first"
+     * instance, the "then" instance must not be allowed to come up.
+     */
+    } else if (pcmk_any_flags_set(type, pe_order_runnable_left
+                                        |pe_order_implies_then)) {
+        pe_rsc_info(then->rsc,
+                    "Inhibiting %s from being active "
+                    "because there is no %s instance to interleave",
+                    then_instance->id, first->rsc->id);
+        return pcmk__assign_resource(then_instance, NULL, true);
```

Comment:
I wonder if `pcmk__unassign_resource()` might work here (note it returns void not bool), since the debug messages in `pcmk__finalize_assignment()` suggest that we failed to assign the resource rather than we're intentionally unassigning it. Haven't looked thoroughly

Reply:
It is a failure to assign; there's an interleaved ordering but no primary resource instance to interleave with, so this is a final condition.